### PR TITLE
Bug 1810629 - Add an Android shortcut to go straight to the login and passwords page

### PR DESCRIPTION
**Pull request by gitstart**
The PR creates a shortcut that users can place on the HomeScreen and when the user clicks the shortcut it will open the `login and passwords page`

### Issue Screenshots
No video this is a new feature not bug

### Fix Screenshots
https://www.loom.com/share/ed2452528b754822afee436b3dca64f8

### Pull Request checklist
 - [ ] Quality: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
 - [ ] Tests: This PR includes thorough tests or an explanation of why it does not
 - [ ] Changelog: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
 - [ ] Accessibility: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features
### After merge
- Milestone: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- Breaking Changes: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

---
This code was written and reviewed by GitStart Community. Growing future engineers, one PR at a time.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1810629

### CODE REVIEW

**Comment by gabrielluong** [comment]
Since the original issue was a feature request, we would need Product to weigh in on this. cc @kbrosnan

**Comment by gitstart** [comment]
> Since the original issue was a feature request, we would need Product to weigh in on this. cc @kbrosnan

Okay @gabrielluong. 🙏
Will wait for the response.

**Comment by amedyne** [comment]
Based on a review today: 
Product decision: This is a useful feature, so we are okay taking in the change.
UX: We need to confirm with Content/Design about the String and menu Lock Icon. FYI @lime124 and @ewachowiak

**Comment by ewachowiak** [comment]
To confirm, the feature works like this:
- User long presses the app icon on Home screen
- User selects "Password manager"
- A shortcut called "Password manager" appears on the Home screen

Once confirmed, I'm going to check with Katie Lower on the strings (she's working on the password feature right now)

**Comment by gitstart** [comment]
> To confirm, the feature works like this:
> 
> * User long presses the app icon on Home screen
> * User selects "Password manager"
> * A shortcut called "Password manager" appears on the Home screen
> 
> Once confirmed, I'm going to check with Katie Lower on the strings (she's working on the password feature right now)

Yes confirmed @ewachowiak

**Comment by ewachowiak** [comment]
Thanks for confirming. The label "Password manager" makes it sound like it will take you to the password manager, not that it will add a shortcut. I suggest changing the text to "Password manager shortcut". I think a bigger issue might be that the feature is not called "password manager" inside Firefox. Let me check with Katie and get back to you. Thanks for your patience.
.

**Comment by ewachowiak** [comment]
Update: I spoke to Katie and confirmed that the long-term plan is to change the name from "Logins and passwords" to just "Passwords." In light of this and of the space restrictions we have here, here is my string recommendation: 
`Passwords shortcut`

**Comment by gitstart** [comment]
> Update: I spoke to Katie and confirmed that the long-term plan is to change the name from "Logins and passwords" to just "Passwords." In light of this and of the space restrictions we have here, here is my string recommendation: `Passwords shortcut`

Thank you @ewachowiak, 
We will make the changes and push shortly.

**Comment by cwzilla** [comment]
Hi, design systems team here. Please see the attachment for the lock icon. Thanks!
lock-24

**Comment by amedyne** [comment]
@gitstart please update with the latest icon from @cwzilla .

**Comment by gitstart** [comment]
> @gitstart please update with the latest icon from @cwzilla .

Hello @amedyne,
Thank you for sharing the icon. We had pushed all the changes.
Please take a look.✔
Thank you.🙌

File: fenix/app/src/main/java/org/mozilla/fenix/home/intent/OpenPasswordManagerIntentProcessor.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+/**
+ * When the open password manager shortcut is tapped, Fenix should open to the password and login fragment.
+ */
+class OpenPasswordManagerIntentProcessor : HomeIntentProcessor {
+
+    override fun process(intent: Intent, navController: NavController, out: Intent): Boolean {
+        val event = intent.extras?.getBoolean(HomeActivity.OPEN_PASSWORD_MANAGER)
+        return if (event != null) {
+            MetricsUtils.Source.SHORTCUT
```
Comment (rocketsroger) [inline fenix/app/src/main/java/org/mozilla/fenix/home/intent/OpenPasswordManagerIntentProcessor.kt:22]: This line is not needed?

**Review by rocketsroger — COMMENTED** [review]
Great change.  Looks good.  One small issue.
I did ran into some issue starting Fenix until I rebase your changes to main.  Please rebase to main and I'll retry to confirm there's no issue.  Thanks

File: fenix/app/src/nightly/res/xml/shortcuts.xml [hunk 2]
```diff
@@ -4,6 +4,17 @@
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
 
 <shortcuts xmlns:android="http://schemas.android.com/apk/res/android">
+    <shortcut
+        android:shortcutId="password_manager"
+        android:enabled="true"
+        android:icon="@drawable/ic_static_shortcut_password_manager"
```
Comment (zmckenney) [inline fenix/app/src/nightly/res/xml/shortcuts.xml:10]: Same issue here. This should be `android:icon="@drawable/ic_static_password_shortcut"`

File: fenix/app/src/release/res/xml/shortcuts.xml [hunk 3]
```diff
@@ -3,6 +3,17 @@
    - License, v. 2.0. If a copy of the MPL was not distributed with this
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
 <shortcuts xmlns:android="http://schemas.android.com/apk/res/android">
+    <shortcut
+        android:shortcutId="password_manager"
+        android:enabled="true"
+        android:icon="@drawable/ic_static_shortcut_password_manager"
```
Comment (zmckenney) [inline fenix/app/src/release/res/xml/shortcuts.xml:9]: Same issue here. This should be `android:icon="@drawable/ic_static_password_shortcut"`

File: fenix/app/src/beta/res/xml/shortcuts.xml [hunk 4]
```diff
@@ -3,6 +3,17 @@
    - License, v. 2.0. If a copy of the MPL was not distributed with this
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
 <shortcuts xmlns:android="http://schemas.android.com/apk/res/android">
+    <shortcut
+        android:shortcutId="password_manager"
+        android:enabled="true"
+        android:icon="@drawable/ic_static_shortcut_password_manager"
```
Comment (zmckenney) [inline fenix/app/src/beta/res/xml/shortcuts.xml:9]: This path is incorrect. This should be `android:icon="@drawable/ic_static_password_shortcut"`

**Review by zmckenney — CHANGES_REQUESTED** [review]

**Review by rocketsroger — APPROVED** [review]
LGTM 🚢

**Review by zmckenney — APPROVED** [review]

**Pull request merged**